x-pack/plugins/infra/server/lib/alerting/log_threshold/log_threshold_executor.ts

```diff
@@ -383,6 +417,7 @@ export const processGroupByResults = (
             group: group.name,
             isRatio: false,
             reason: reasonMessage,
+            viewInAppUrl,
```

Comment:
Could this be baked into the `alertFactory` or `alertUpdater` directly, which is already passed down? Then we wouldn't need to thread through two additional arguments.

Reply:
@weltenwort, I'm not too fond of passing two additional params for each function. And I was thinking of putting the `viewInAppUrl` inside https://github.com/fkanout/kibana/blob/eae32157d873bc604b00ee618624cd44e4b8ee06/x-pack/plugins/infra/server/lib/alerting/log_threshold/log_threshold_executor.ts#L489. But I went with the current solution because I didn't know if I could pack any additional arguments into the functions you mentioned, who are coming from the rule registry in the first place. Let me see what I can do about that.

Reply:
Done

Reply:
Looks like we're still passing the url through instead of creating an updater bound to it. Couldn't we do that the same way we pre-bind the fields in the `alertFactory`?

Reply:
I'm not sure what do you mean by pre-bind? Do you mean build `viewInAppUrl` inside the `alertFactory` then retune it alongside `alertWithLifecycle` within an object? It doesn't make sense to me from a separating of concerns perspective.

It is worth mentioning too, that we need the `viewInAppUrl` when the `alertUpdater` function is called.

Reply:
Done